README.md
# Xsolla Inc. Unity SDK VERSION 1.3.7

### GETTING STARTED

1. To become an Xsolla partner, sign up in Xsolla control panel at this address: https://publisher.xsolla.com/signup/?utm_source=unity&utm_medium=instruction  
2. Familiarize yourself with the Integration Guides and select the modules that best fit your project. You can find the Integration Guides here: https://developers.xsolla.com/#getting-started


**Features:**  
 - Saved payment methods;
 - Purchase of virtual currency;
 - Purchase of virtual items;
 - Purchase of subscriptions;
 - Promotions;
 - Redeem coupon;
 - User's payment history.
 - Saved payment methods management
 - Subscription management

• Set Up Instant Payment Notification (IPN): http://developers.xsolla.com/api.html#notifications  
• Create an Xsolla Access Token to conduct payments with maximum security. You can find documentation on creating a token here: http://developers.xsolla.com/api.html#payment-ui


##### Get Started

If you would like to accept payments through Xsolla’s payment UI, follow these steps:  
 1. Create empty GameObject on your game scen and add **XsollaSDK.cs** script on them. Or add script on any other object.
 2. To create Payment form you need call  
```cs
    CreatePaymentForm(token, actionOk(XsollaOkResult), actionError(XsollaError))  
```

|Member   | Description|
|------   | -----------|
|token    | Your purchase token ([Getting token method][3508ac7b])
|actionOk | Call when payment process completed, delegate here your func.<br>*Example: OnResulOkReceivied(XsollaOkResult result){Debug.Log(“Ok”);}* |
|actionError | Call when payment process canceled or some problems appeared, delegate here your func.<br>*Example: OnErrorReceivied(XsollaError data){Debug.Log(“Error”);}*|

[3508ac7b]: https://developers.xsolla.com/api.html#token "Getting token method"

 - Also you can use XsollaSDK.InitPaystation(string token) function to use our payment solution in native browser.

#### SDK RESPONSE OBJECTS

```cs
public class XsollaResult {
    public string invoice{ get; set;}
    public Status status{ get; set;}
    // DONE status mean successful payment
    public Dictionary<string, object> purchases;
    // purchases can contain
    // Key: «out» - virtual currency      | Value: int amount
    // Key: «id_package» - subscription   | Value: long subscriptionId
    // Key: «sku[itemId]» - virtual items | Value: int amount
}
```
```cs
public class XsollaError {
    public Source errorSource{ get; private set;}
    public int errorCode{ get; private set;}
    public string errorMessage{ get; private set;}
}
```

#### TRY IT!


You can look demo on https://livedemo.xsolla.com/sdk/unity/  
We have two test scenes in "XsollaUnitySDK" -> "Resources" -> "_Scenes" folder:  
- XsollaFarmFreshScene - emulates item shop  
- XollaTokenTestScene - here you can test your token.


CHANGELOG.md
# Changelog

## 1.3.7 - 2017-04-14

  * Shop screens design update
  * Grid/list shop items view support added
  * Light theme support added
  * Serverless integration support added
  * High resolution screen support added
  * Images rendering quality improved
  * Work speed improved

## 1.3.6 - 2017-02-17

The Subscription Management Interface was released.
Main Features:
  * List of active subscriptions
  * Subscription renewal
  * Stop subscription renewal
  * Subscription canceling
  * Deleting of the payment method account linked to subscription
  * Linking of the payment method account for subscription renewal
  * History of subscription payments

## 1.3.5 - 2017-01-24
- [update] add functional «Payment manager»

## 1.3.4 - 2016-12-27

- [update] add functional "Select Subscription"
- [fix] minor text fixes

## 1.3.3 - 2016-10-21

- [update] add functional «Profile History»
- [update] add functional «Custom Virtual Currency Amount»
- [fix] minor text fixes

## 1.3.2 - 2016-09-30

- [update] add functional «Redeems coupon»
- [update] add functional «Gift card»
- [update] add functional «Promo code»
- [update] some design update
- [fix] minor text fixes

## 1.3.1 - 2016-08-24

- [update] add work with saved payment methods  
- [update] some design update
- [fix] minor text fixes

## 1.3.0 - 2016-07-26

- [update] add TLS support > 1.0


Scripts/Api/XsollaApiConst.cs

namespace Xsolla {
    public struct XsollaApiConst {
		public const string ERROR_CODE = "error_code";
		public const string ERROR_MSG = "error";
		public const string REQUEST_PARAMS = "request_params";

		public const string COMMAND_CALCULATE = "calculate";
		public const string COMMAND_CHECK = "check";

		//XPS params 
		public const string XPS_FIX_COMMAND = "xps_fix_command";
		public const string XPS_CHANGE_ELEM = "xps_change_element";

        //form fields
		public const string ACCESS_TOKEN = "access_token";
        public const string PID = "pid";
        public const string OUT = "out";
        public const string SKU = "sku";
        public const string ID_PACKAGE = "id_package";
        public const string RETURN_URL = "returnUrl";
        public const string PAYMENT_WITH_SAVED_METHOD = "paymentWithSavedMethod";

        public const string XPS = "xps";
        public const string DESCRIPTION = "description";
        public const string CARD_NUMBER = "card_number";
        public const string CARD_EXP_YEAR = "card_year";
        public const string CARD_EXP_MONTH = "card_month";
        public const string CARD_CVV = "cvv";
        public const string CARD_ZIP = "zip";
        public const string CARD_HOLDER = "cardholdername";
        public const string ALLOW_SUBSCRIPTION = "allowSubscription";
		public const string COUPON_CODE = "couponCode";

        //PAYMENT SYSTEM ID
        public const string PID_Card = "26";
        public const string PID_PayPal = "24";

        //R - RESPONSE
        public const string R_TITLE = "title";
        public const string R_TEXTALL = "textAll";
        public const string R_ERRORS = "errors";
		public const string R_MESSAGES = "messages";
        public const string R_ACCOUNT = "account";
        public const string R_ACCOUNTXSOLLA = "accountXsolla";
        public const string R_ONLINEBANKING = "onlineBanking";
        public const string R_FORM = "form";
        public const string RF_DFP = "dfp";
        public const string R_CHECKOUTFORM = "checkoutForm";
        public const string R_JAVASCRIPT = "javascript";
        public const string R_REPORT = "report";
        public const string R_BACKURL = "backURL";
        public const string R_BACKVISIBLE = "backVisible";
        public const string R_FORMURL = "formURL";
        public const string R_SKIPFORM = "skipForm";
		public const string R_SKIPCHECKOUT = "skipCheckout";
        public const string R_FORMPARAMS = "formParams";
        public const string R_FORMISHIDDEN = "formIsHidden";
        public const string R_INSTRUCTION = "instruction";
        public const string R_BUYDATA = "buyData";
        public const string R_CURRENCY = "currency";
        public const string R_VIRTUALCURRENCYNAME = "virtualCurrencyName";
        public const string R_INSTANCESIZE = "instanceSize";
        public const string R_STATUS = "status";
        public const string R_MARKETPLACE = "marketplace";
        public const string R_HASFORMBEENSUBMITTED = "hasFormBeenSubmitted";
        public const string R_CURRENTCOMMAND = "currentCommand";
        public const string R_MINMAX = "minmax";
        public const string R_REQUESTID = "requestID";
        public const string R_SCHEMEREDIRECT = "schemeRedirect";
        public const string R_PID = "pid";
        public const string R_ICONURL = "iconUrl";
        public const string R_ACTION = "action";
        public const string R_INFOFIELDS = "infofields";
        public const string R_PROJECTLOGO = "projectLogo";
        public const string R_FORMCLASS = "formClass";
        public const string R_PROJECT = "project";
        public const string R_THEME = "theme";
        public const string R_REQUEST = "request";
        public const string R_COUNTRY = "country";
        public const string R_APP = "app";
        public const string R_CONTEXT = "context";
        public const string R_ISPACKETS = "isPackets";
        public const string R_ISSHORTFORM = "isShortForm";
        public const string R_ISFASTCHECKOUT = "isFastCheckout";
        public const string R_SPIKES = "spikes";
		public const string R_USER = "user";
		public const string R_PURCHASE = "purchase";
		public const string R_SETTINGS = "settings";
		public const string R_TRANSLATIONS = "translations";
		public const string R_API = "api";
		public const string USER_INITIAL_CURRENCY = "userInitialCurrency";
    }
}
